chia/wallet/wallet_state_manager.py

```diff
@@ -966,6 +1048,71 @@ async def handle_nft(
             wallet_identifier = WalletIdentifier.create(new_nft_wallet)
         return wallet_identifier
 
+    async def handle_clawback(
+        self,
+        clawback_args: Tuple[uint64, bytes32, bytes32],
+        coin_state: CoinState,
+        peer: WSChiaConnection,
+    ) -> Optional[WalletIdentifier]:
+        """
+        Handle Clawback coins
+        :param clawback_args: uncurried clawback parameters
+        :param coin_state: clawback merkle coin
+        :param peer: Fullnode peer
+        :return:
+        """
+        time_lock, sender_puzhash, recipient_puzhash = clawback_args
+        # Record metadata
+        metadata = {
+            "time_lock": int(time_lock),
+            "recipient_puzhash": recipient_puzhash.hex(),
+            "sender_puzhash": sender_puzhash.hex(),
+        }
+        assert coin_state.created_height is not None
+
+        # Check if the wallet is the sender
+        sender_derivation_record: Optional[
+            DerivationRecord
+        ] = await self.puzzle_store.get_derivation_record_for_puzzle_hash(sender_puzhash)
+        if sender_derivation_record is not None:
+            self.log.info("Found Clawback merkle coin %s as the sender.", coin_state.coin.name().hex())
+            metadata["is_recipient"] = False
+            coin_record = WalletCoinRecord(
+                coin_state.coin,
+                uint32(coin_state.created_height),
+                uint32(0),
+                False,
+                False,
+                WalletType.STANDARD_WALLET,
+                1,
+                CoinType.CLAWBACK,
+                json.dumps(metadata),
+            )
+            # Add merkle coin
+            await self.coin_store.add_coin_record(coin_record)
```

Comment:
Yes, it's very useful to return the coin type since we need to handle different types of coins after they are spent. We are stopping creating derivation paths for the new type of wallets. So their coins cannot be marked as spent in a common way.